Source/Core/DolphinWX/Input/InputConfigDiag.cpp

```diff
@@ -342,17 +342,12 @@ void ControlDialog::UpdateGUI()
   m_bound_label->SetLabel(
       wxString::Format(_("Bound Controls: %lu"), (unsigned long)control_reference->BoundCount()));
 
-  switch (control_reference->GetParseStatus())
-  {
-  case ParseStatus::SyntaxError:
+  if (control_reference->GetParseStatus() == ParseStatus::SyntaxError)
     m_error_label->SetLabel(_("Syntax error"));
-    break;
-  case ParseStatus::NoDevice:
+  else if (control_reference->BoundCount() == 0)
```

Comment:
A problem with this change is that Dolphin can't correctly tell whether the device is not found. If you just have nothing bound to the input, it will still show "Device not found"

Reply:
@leoetlino I agree that when there's no control or device bound, it shouldn't show that message.

What's the correct behavior here if you have a control bound, but no device? For example, I could assign the letter 'X' on my keyboard to the A button, then clear the device dropdown (so it says "/0/"). That sort of seems like a "device not found" situation to me, but maybe it's more like "device not specified"?

Reply:
IIRC, master shows "device not found" when the device cannot be found or no device has been specified, so I'm fine with either.

Reply:
Okay, I added a new enum for `ParseStatus::EmptyExpression`, which is returned when it can't possibly bind to anything (i.e., it's an empty or all-whitespace string). InputConfigDiag handles it by setting an empty error message.